# scRNA-Seq-Analysis

## Introduction to RNA seq
Used to measure gene expression.

### Three main steps for RNA seq
1. Prepare a sequence library
   - Isolate the RNA
     
   - Break RNA into small fragments
     
   - Convert RNA into dsDNA
     
   - Add adapters
     
   - PCR amplification
2. Sequence
   - Sequencer/sequencing machine
     
   - Quality control
     
3. Data analysis

<img width="500" alt="scrnaseq2" src="https://github.com/user-attachments/assets/690089f1-c845-4fd5-8c22-df585aadf375">

### BULK
- Extract RNA
  
- Average gene expression across all the cells
  
- Impossible to compare differences in gene expression between individual cells
  

### SINGLE
- Extract RNA from all the cells
  
- Sequence RNA from each cell and quantify the expression

## Bulk v/s single-cell RNA sequencing

SINGLE-CELL RNA SEQ ADVANTAGES

- Allows you to compare the expression between cells
  
- Identify rare cell population
  
- Understand spatial transcriptomics
  
- Track cell lineage

# Basic Workflow

![image](https://github.com/user-attachments/assets/69498924-0392-4acb-b250-ad179e0d9fa0)

<img width="600" alt="scRNA3" src="https://github.com/user-attachments/assets/66e3a977-222b-49ca-9233-225064ca6631">

### Single-cell Isolation

- Limiting Dilution - Use pipettes to isolate cells by dilution
  
- Micromanipulation - Classic approach used to retrieve cells from samples with a small number of cells such as early embryos
  
- Flow-activated cell sorting - widely used for isolating single cells in suspension
  
- Fraser capture microdissection - An advanced technique used for isolating single cells from solid tissue by using a laser system


